docs/ca/dev/monitoring/opentelemetry-instrumentation.md

```diff
@@ -0,0 +1,446 @@
+We would recommend to do so using `MonitoringService::addCustomParameter()` method. By default Spryker provides a few of those attributes already, like a current store name and locale.
+If you don't want to use `MonitoringService` for some reason, you may use `\Spryker\Service\Opentelemetry\OpentelemetryService::setCustomParameter` directly. It does exactly the same job, but without necessity to install `Monitoring` module.
+All attributes added via those services will be added into the root span on the very end of request execution, so you can call it even after response wes already sent.
+
+## Custom events
+
+If you want to add some custom event to you trace in order to configure some logic into your backend, you can trigger a custom event during the execution. It works pretty much the same as adding custom attributes, so it's pretty easy to use.
+The same as for custom attributes you can add those via `\Spryker\Service\Monitoring\MonitoringService::addCustomEvent()` or via `\Spryker\Service\Opentelemetry\OpentelemetryService::addEvent()` if you dont want to install `Monitoring` module.
+Event will be wired into the root span.
+
+## Error handling
+
+You are definitely want to track all possible errors during the execution. OOTB OpenTelemetry integration will catch all exceptions that were thrown during request/command execution and attach those as events into the root span. Those events should also be present in the span of the method that thrown the exception, but only if we have a hook for this method in the first place.
+In order to catch those error in any case we would recommend to use `\Spryker\Service\Monitoring\MonitoringService::setError()` or `\Spryker\Service\Opentelemetry\OpentelemetryService::setError()` inside of the Error Handler of your application.
+Default Spryker's Error Handler already executes `\Spryker\Service\Monitoring\MonitoringService::setError()`, so if you are using `Monitoring` module and default Error Handler - you are covered. But if you don't, please adjust your error handler accordingly.
+This will add a error event into the root span and will change its status to the `error` one. We would highly recommend to check this part during integration of OpenTelemetry into your system.
+
+## Changing a service name
+
+Service name allows you to filter traces by the source. You may want to check only Yves traces or mb just to check only Glue requests, but you backend can show you a bunch of CLI commands from your Scheduler.
+
+You can define a service name via `\Spryker\Service\Monitoring\MonitoringService::setApplicationName()` or `\Spryker\Service\Opentelemetry\OpentelemetryService::setResourceName()`.
+
+If you are using the `MonitoringService` and any method from it was executed, you will get a generated service name in any case.
```

Comment:
This is not clear

Reply:
MonitoringService has a bunch of methods and each execution of any of it will trigger method that sets service name for monitoring tool